frontend/components/forms/admin/AppConfigForm/AppConfigForm.tsx

```diff
@@ -160,14 +163,27 @@ const AppConfigFormFunctional = ({
       errors.org_name = "Organization name must be present";
     }
 
+    if (orgLogoURL && !validUrl(orgLogoURL)) {
+      errors.org_logo_url = `${orgLogoURL} is not a valid URL`;
+    }
+
     if (!serverURL) {
       errors.server_url = "Fleet server URL must be present";
     }
 
     if (enableSSO) {
-      if (metadata === "" && metadataURL === "") {
-        errors.metadata_url = "Metadata URL must be present";
+      if (idpImageURL && !validUrl(idpImageURL)) {
+        errors.idp_image_url = `${idpImageURL} is not a valid URL`;
+      }
+
+      if (metadata === "") {
+        if (metadataURL === "") {
+          errors.metadata_url = "Metadata URL must be present";
+        } else if (!validUrl(metadataURL)) {
+          errors.metadata_url = `${metadataURL} is not a valid URL`;
+        }
```

Comment:
I realize we can't do the `return` pattern here because we need it to finish parsing the function. But, it's a good example of why we need another pattern for this validation. It's getting pretty difficult to maintain with all of the nested if's and else if's.